lib/collection/variable-list.js

```diff
@@ -180,19 +240,32 @@ _.extend(VariableList, /** @lends VariableList */ {
      *
      * @param {Object|Array} items
      * @returns {Object}
+     *
+     * @todo maybe move to instance methods or take instance as parameter? will help to create list specific keys and
+     * case senitivity
      */
     objectify: function (items) {
         // convert array to an indexed object
-        _.isArray(items) && (items = _.indexBy(items, 'id')); // in any case variable without `id` is useless
-        // create an object with `id` as keys and an instance of `Variable` as their value.
+        // in any case variable without `key` is useless
+        _.isArray(items) && (items = _.indexBy(items, Variable._postman_propertyIndexKey));
+        // create an object with `key` as keys and an instance of `Variable` as their value.
         return _.mapValues(items, function (item, key) {
             // if already an instance of `Variable`, we simply return it, else we create a new instance of `Variable`
-            return Variable.isVariable(item) ? item : (new Variable((_.isPlainObject(item) && _.has(item, 'value')) ?
-                item : {
-                    id: key,
-                    value: item
-                }
-            ));
+            if (Variable.isVariable(item)) {
+                return item;
+            }
+
+            // if the item is a valid object, we use the entire object as a variable definition
+            // also works to handle array type definition along with object type definition
+            var obj = item;
+
+            // in case item is not a valid object, we create our own object + it handles object-type definition
+            if (!(_.isPlainObject(obj) && _.has(obj, 'value'))) {
+                obj = { value: item };
```

Comment:
There is a negation in the if block. If the said input is not a plain object or does not have value property, what you mentioned is right.